src/sparsezoo/api/query_parser.py

```diff
@@ -93,12 +93,64 @@ def _parse_arguments(self) -> None:
 
     def _parse_fields(self) -> None:
         fields = self.fields or DEFAULT_FIELDS.get(self.operation_body)
-
+        if isinstance(fields, List):
+            self.fields = self.parse_list_fields_to_string(fields)
+        elif isinstance(fields, Dict):
+            """
+            fields = {
+                "model_id": None,
+                "benchmarks": {
+                    benchmark_result_id: None,
+                    whatever: {
+                        ...
+                    }
+                }
+            }
+            """
+            self.fields = self.parse_dict_fields_to_string(fields)
+
+    def parse_list_fields_to_string(self, fields: List[str]) -> str:
         parsed_fields = ""
         for field in fields:
             camel_case_field = to_camel_case(field)
             parsed_fields += f"{camel_case_field} "
-        self.fields = parsed_fields
+            if camel_case_field in DEFAULT_FIELDS:
+                parsed_fields += (
+                    "{ "
+                    + self.parse_list_fields_to_string(
+                        DEFAULT_FIELDS.get(camel_case_field)
+                    )
+                    + "} "
+                )
+        return parsed_fields
+
+    def parse_dict_fields_to_string(self, fields: Dict[str, Optional[Dict]]) -> str:
+        parsed_fields = ""
+        for field, field_dict in fields.items():
+            if field_dict:
+                assert isinstance(field_dict, Dict), (
+                    "QueryParser fields must be either dict where ",
+                    "values are a dict or None",
+                )
+
+                parent_field = to_camel_case(field)
+                child_fields = (
+                    " { " + self.parse_dict_fields_to_string(field_dict) + "} "
+                )
+                parsed_fields += parent_field + child_fields
+            else:
+                camel_case_field = to_camel_case(field)
+                parsed_fields += f"{camel_case_field} "
+                if camel_case_field in DEFAULT_FIELDS:
+                    parsed_fields += (
+                        "{ "
+                        + self.parse_list_fields_to_string(
+                            DEFAULT_FIELDS.get(camel_case_field)
+                        )
+                        + "} "
+                    )
```

Comment:
This is the same logic from above, no? Maybe a helper method would help DRY things? Not sure if there's value in that, but worth considering in this case I'd think.